src/d3d11/d3d11_texture.cpp

```diff
@@ -38,6 +41,17 @@ namespace dxvk {
     imageInfo.initialLayout   = VK_IMAGE_LAYOUT_UNDEFINED;
     imageInfo.shared          = vkImage != VK_NULL_HANDLE;
 
+    if (m_desc.MiscFlags & (D3D11_RESOURCE_MISC_SHARED|D3D11_RESOURCE_MISC_SHARED_NTHANDLE)) {
```

Comment:
We should always keep shared images in `GENERAL` layout as well, so the `OptimizeLayout` call below needs to be conditional on the image not being shared.